mlflow/deployments/__init__.py

```diff
@@ -13,14 +13,20 @@
 your own plugin for deployment to a custom serving tool, see
 `plugin docs <http://mlflow.org/docs/latest/plugins.html#writing-your-own-mlflow-plugins>`_.
 """
+import contextlib
 import json
 
 from mlflow.deployments.base import BaseDeploymentClient
-from mlflow.deployments.databricks import DatabricksDeploymentClient  # noqa: F401
+from mlflow.deployments.databricks import DatabricksDeploymentClient
 from mlflow.deployments.interface import get_deploy_client, run_local
 from mlflow.exceptions import MlflowException
 from mlflow.protos.databricks_pb2 import INVALID_PARAMETER_VALUE
 
+with contextlib.suppress(Exception):
+    # MLflowDeploymentClient depends on optional dependencies and can't be imported
+    # if they are not installed.
+    from mlflow.deployments.mlflow import MLflowDeploymentClient
```

Comment:
@harupy does it make more sense to catch `ImportError` here?

Reply:
Good point. I think `Exception` is safer because there is no guarantee that our pydantic models will be compatible with a future version of pydantic.

https://github.com/mlflow/mlflow/pull/9337

Reply:
Nice - sounds good!